types/index.d.ts

```diff
@@ -349,11 +355,11 @@ interface Knex<TRecord extends {} = any, TResult = unknown[]>
   transaction(
     transactionScope?: null,
     config?: any
-  ): Promise<Knex.Transaction>;
+  ): Promise<Knex.Transaction> & TransactionMethods<Promise<Knex.Transaction>>;
```

Comment:
Shouldn't this also be added to TransactionProvider?

Reply:
Uhhh I haven't touched the TransactionProvider, so I'm not sure if it's the same code path, but sure, I could add it there